### PR TITLE
Fix false positives with async closures for large_tuple rule

### LINKED ISSUE #3753: large_tuple violation false positive when async function is a method parameter
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
When a function accepts an `async` function of 3+ parameters as a parameter, it is erroneously considered a violation of the `large_tuple` rule.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ echo "private func foo(bar: (Int, String, Float) async throws -> Void) {}" | swiftlint lint --no-cache --use-stdin --enable-all-rules
```

Output:
```
<nopath>:1:18: warning: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.45.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Built from source.
* Paste your configuration file: No config file, this reproduces with the default configuration options.

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcodebuild -version`)? I'm on Linux, so no Xcode version.
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
private func foo(bar: (Int, String, Float) async throws -> Void) {}
```

### PR DESCRIPTION
**Pull request by kmahar**
Fixes #3753. Will leave some comments inline

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Metrics/LargeTupleRuleExamples.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
+struct LargeTupleRuleExamples {
```
Comment (kmahar) [inline Source/SwiftLintFramework/Rules/Metrics/LargeTupleRuleExamples.swift:1]: pulling out the examples into a separate file like this seemed to a strategy used elsewhere in the codebase to add Swift version-specific examples, as seen in `ImplicitGetterRuleExamples.swift`, but let me know if there some better way to handle this.

File: Source/SwiftLintFramework/Rules/Metrics/LargeTupleRuleExamples.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            Example("var completionHandler: ((_ data: Data?, _ resp: URLResponse?, _ e: NSError?) -> Void)!\n"),
+            Example("func getDictionaryAndInt() -> (Dictionary<Int, String>, Int)?\n"),
+            Example("func getGenericTypeAndInt() -> (Type<Int, String, Float>, Int)?\n")
+        ]
+
+        guard SwiftVersion.current >= .fiveDotFive else {
+            return commonExamples
+        }
+
+        return commonExamples + [
```
Comment (kmahar) [inline Source/SwiftLintFramework/Rules/Metrics/LargeTupleRuleExamples.swift:26]: For both the triggering and non-triggering examples, I just copied the existing ones that involved functions and added `async`. Not all of these were actually false positives -- the issue is only with async closures -- but given that both `throws` and non-`throws` versions were tested for all of the functions already it seemed like getting coverage of the `async` versions would be a good idea too.

**Comment by kmahar** [comment]
(happy to add additional tests if desired per the bot comment, but the only existing ones I found for the rule for those auto generated from the triggering/non-triggering examples.)

**Comment by jpsim** [comment]
Thanks for the PR @kmahar! The Danger bot doesn't consider SwiftLint's automatically-generated tests, but the examples you added are the right way to test this change.

I'll merge shortly.

**Pull request merged**